Source/WebCore/dom/ViewTransition.cpp

```diff
@@ -825,7 +822,21 @@ void ViewTransition::clearViewTransition()
         documentElement->invalidateStyleInternal();
 }
 
-Ref<MutableStyleProperties> ViewTransition::copyElementBaseProperties(RenderLayerModelObject& renderer, LayoutSize& size, LayoutRect& overflowRect, bool& intersectsViewport)
+// Rounds the x/y translation components to the nearest pixel (for non-perspective)
+// transforms, and returns the subpixel offset that was removed.
+static LayoutSize snapTransformationTranslationToDevicePixels(TransformationMatrix& matrix, float deviceScaleFactor)
+{
+    if (matrix.hasPerspective())
+        return { };
+
+    LayoutSize oldTranslation(matrix.m41(), matrix.m42());
+    matrix.setM41(std::round(matrix.m41() * deviceScaleFactor) / deviceScaleFactor);
+    matrix.setM42(std::round(matrix.m42() * deviceScaleFactor) / deviceScaleFactor);
+    return oldTranslation - LayoutSize(matrix.m41(), matrix.m42());
+}
+
+
+void ViewTransition::copyElementBaseProperties(RenderLayerModelObject& renderer, CapturedElement::CaptureState& output)
```

Comment:
```suggestion
void ViewTransition::copyElementBaseProperties(RenderLayerModelObject& renderer, CapturedElement::State& output)
```

This would read slightly nicer now that I see the usages. I'd just rename to `State` as `CapturedElement` provides enough context